Enable CORS and using parameters after ? in get command

MPrtenjak opened the pull request:
1.	Enabling CORS (Cross-origin resource sharing)
By enabling field ($server->useCors = true) REST server can be used for Cross-origin resource sharing which is useful if you need to access a server from development computer

2.	Enable using GET parameters after ?
Sometime it is useful to have parameters in GET request (aka: http://something.com/users?active=1 – select only active users). In this case parameter active can be obtained with: $server->getParamAssoc['active']

tyzoid commented:
Sorry for the delay. Thanks for implementing this, we've received some requests for the CORS feature for some time. I'm going to do some testing, then I'll see where we're at.

tyzoid commented on source/Jacwright/RestServer/RestServer.php:
```diff
@@ -535,7 +550,13 @@ private function xml_encode($mixed, $domElement = null, $DOMDocument = null) {
 	}
 
-
+  private function corsHeaders() {
```
Indentation here seems off.

tyzoid commented on source/Jacwright/RestServer/RestServer.php:
```diff
@@ -372,6 +380,9 @@ protected function generateMap($class, $basePath) {
 
 	public function getPath() {
+		$getParameters = parse_url ($_SERVER['REQUEST_URI'], PHP_URL_QUERY);
+		parse_str($getParameters, $this->getParamAssoc);
```
Not sure why this is necessary, I'll do some tests to double-check, but iirc `$_GET` worked fine.

tyzoid commented on source/Jacwright/RestServer/RestServer.php:
```diff
@@ -111,6 +113,11 @@ public function handle() {
+			$this->corsHeaders();
+			exit;      
+		}
+		
```
Careful not to add extra whitespace at the end of the lines here.

MPrtenjak commented:
Hi,

I have fixed indentation and removed unnecessary code, but I don’t know what I must do next. Do I open a new pull request or how to merge new code – sorry I am no familiar with GitHub.

--

And now about parameters after ‘?’. I have removed the code. To be honest I don’t know how to get those without my ‘Enhancement’. There are no examples in your code how to achieve that. (PS: PHP is not my preferred language and I am using it because I (currently) have no other choice)


Thanks for good job and have a nice day.

tyzoid commented:
Nope, I'll handle the merge. As for how to access these parameters, try this controller:

    <?php
    class GetParams {
        /**
         * @name DumpGet
         * @description Dumps the get superglobal
         *
         * @url GET /dumpGet
         */
        public function dumpGet() {
            return array('$_GET' => $_GET);
        }
    };

MPrtenjak commented:
`return array('$_GET' => $_GET);` works great

Thanks!

tyzoid commented:
Glad that worked out for you.
